src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -36,10 +36,13 @@ ClusterTaskManager::ClusterTaskManager(
       get_node_info_(get_node_info),
       announce_infeasible_task_(announce_infeasible_task),
       local_task_manager_(std::move(local_task_manager)),
-      scheduler_resource_reporter_(
-          tasks_to_schedule_, infeasible_tasks_, *local_task_manager_),
-      internal_stats_(*this, *local_task_manager_),
-      get_time_ms_(get_time_ms) {}
+      get_time_ms_(get_time_ms) {
+  if (local_task_manager_) {
+    scheduler_resource_reporter_ = std::make_unique<SchedulerResourceReporter>(
```

Comment:
Even without local_task_manager (gcs case), we still need to report infeasible_tasks since it's in the cluster_task_manager

Reply:
Ditto.